sbi/utils/user_input_checks.py

```diff
@@ -585,23 +588,21 @@ def process_x(
     x = atleast_2d(torch.as_tensor(x, dtype=float32))
 
     # If x_shape is provided, we can fix a missing batch dim for >1D data.
-    if x_shape is not None and len(x_shape) > len(x.shape):
+    if x_event_shape is not None and len(x_event_shape) == len(x.shape):
```

Comment:
Why is this necessary?
Should this not raise an error if len(x_event_shape) > len(x.shape). I guess this will be done anyway by the DensityEstimator.

Reply:
It adds a batch dimension to `x`.

For multi-dimensional data (images) the line above `x = atleast_2d(...)` does not add the batch dimension yet.

Reply:
But I agree we can raise here, I will add it.